### PR TITLE
perf: avoid heap allocation for getting connections

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

avoid heap allocation for getting connections

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Comment by ahabhgk** [comment]
!bench

File: crates/rspack_plugin_warn_sensitive_module/src/lib.rs [hunk 1]
```diff
@@ -27,17 +27,18 @@ impl WarnCaseSensitiveModulesPlugin {
 
     for m in modules {
       if let Some(boxed_m) = graph.module_by_identifier(&m) {
-        let mut module_msg = format!("  - {}\n", m);
+        message.push_str("  - ");
+        message.push_str(&m);
+        message.push('\n');
```
Comment (JSerFeng) [inline crates/rspack_plugin_warn_sensitive_module/src/lib.rs:32]: Is that more performant ?

**Review by JSerFeng — APPROVED** [review]

**Comment by JSerFeng** [comment]
For real ? 
image
I didn't realize clone costs that much

**Comment by chenjiahan** [comment]
@JSerFeng I think it's related to https://github.com/web-infra-dev/rspack/pull/8620

**Pull request merged**